ext/zip/php_zip.c

```diff
@@ -1519,21 +1519,21 @@ static ZIPARCHIVE_METHOD(setPassword)
 	char *password;
 	size_t	password_len;
 
-	ZIP_FROM_OBJECT(intern, self);
-
 	if (zend_parse_parameters(ZEND_NUM_ARGS(), "s", &password, &password_len) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	if (password_len < 1) {
 		RETURN_FALSE;
+	}
+
+	ZIP_FROM_OBJECT(intern, self);
```

Comment:
I would suggest to perform this call always directly after ZPP, to keep behavior a bit more consistent across functions.

Reply:
Ahh, yes, I wanted to ask this, but forgot it.